atlasdb-dbkvs/src/main/java/com/palantir/atlasdb/keyvalue/dbkvs/OracleTableNameUnmapper.java

```diff
@@ -99,33 +102,49 @@ private Set<String> getBatchOfLongTableNamesFromMappingTable(
 
         // We use the Oracle LOWER function rather than mapping to lower case on the client to ensure that we're
         // using the same locale and conversion.
-        AgnosticResultSet results = conn.selectResultSetUnregisteredQuery(
-                "SELECT table_name FROM " + AtlasDbConstants.ORACLE_NAME_MAPPING_TABLE + " WHERE"
-                        + " LOWER(short_table_name) IN (" + placeHolders + ")",
-                shortTableNames.toArray());
+        Map<String, String> shortTableToLongTableMapping = runIgnoringTableNotFound(
```

Comment:
Does mean we do use exceptions for control flow :/ However, polling for the existence of the table can lead to a race condition and an extra DB call, so I made the exception (pun not intended) here.